src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,152 @@
+/* eslint-disable max-len */
+import { useEffect, useState } from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { getSearchWith } from '../utils/searchHelper';
 
 export const PeoplePage = () => {
-  return (
-    <>
-      <h1 className="title">People Page</h1>
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState('');
+  const [searchParams, setSearchParams] = useSearchParams();
 
-      <div className="block">
-        <div className="columns is-desktop is-flex-direction-row-reverse">
-          <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
-          </div>
+  const query = searchParams.get('query') || '';
+  const filterBySex = searchParams.get('sex') || null;
+  const centuriesOption = searchParams.getAll('centuries') || [];
+  const sortByUrl = searchParams.get('sort') || null;
+  const sortByOrder = searchParams.get('order') || null;
+  const { personSlug } = useParams();
+
+  let filteredPeople = [...people];
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const getPeopleFromServer = await getPeople();
+
+        const peopleFromServer = getPeopleFromServer.map(person => {
+          const mother = getPeopleFromServer.find(p => p.name === person.motherName);
+          const father = getPeopleFromServer.find(p => p.name === person.fatherName);
+
+          return {
+            ...person,
+            mother,
+            father,
+          };
+        });
+
+        setPeople(peopleFromServer);
+      } catch {
+        setIsError('Something went wrong');
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  if (filterBySex) {
+    filteredPeople = filteredPeople.filter(person => person.sex === filterBySex);
+  }
+
+  useEffect(() => {
+    if (!query) {
+      const updatedSearchParams = getSearchWith(searchParams, { query: null });
 
-          <div className="column">
-            <div className="box table-container">
-              <Loader />
+      setSearchParams(updatedSearchParams);
+    }
+  }, [query, searchParams, setSearchParams]);
 
-              <p data-cy="peopleLoadingError">Something went wrong</p>
+  if (query) {
+    filteredPeople = filteredPeople.filter(person => {
+      return person.name.toLowerCase().includes(query.toLowerCase())
```

Comment:
looks like we can reuse `.toLowerCase().includes(query.toLowerCase())`, let's create separated function for this